### PR TITLE
New package: py-zc-lockfile

### PR DESCRIPTION
**Pull request by marcmengel**
interprocess locking in pythyon, dependency of cherrypy framework

### CODE REVIEW

File: var/spack/repos/builtin/packages/py-zc-lockfile/package.py [hunk 1]
```diff
@@ -0,0 +1,12 @@
+from spack import *
+
+
+class PyZcLockfile(PythonPackage):
+    """Basic inter-process locks """
+
+    homepage = "http://www.python.org/pypi/zc.lockfile"
+    url      = "https://files.pythonhosted.org/packages/58/c2/d7c89bdad237b4b7837609172be3e8bf5630796c0020494a15b97ece8eb1/zc.lockfile-1.4.tar.gz"
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-zc-lockfile/package.py]:
```suggestion
    url      = "https://pypi.io/packages/source/z/zc.lockfile/zc.lockfile-1.4.tar.gz"
```

File: var/spack/repos/builtin/packages/py-zc-lockfile/package.py [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+
+    version('1.4', sha256='95a8e3846937ab2991b61703d6e0251d5abb9604e18412e2714e1b90db173253')
+
+    depends_on('py-setuptools', type='build')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-zc-lockfile/package.py]:
```suggestion
    depends_on('py-setuptools', type=('build', 'run'))
```
I _think_ this one is used at run-time as well.

File: var/spack/repos/builtin/packages/py-zc-lockfile/package.py [hunk 3]
```diff
@@ -0,0 +1,13 @@
+from spack import *
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-zc-lockfile/package.py]: You accidentally deleted the license

File: var/spack/repos/builtin/packages/py-zc-lockfile/package.py [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+    """Basic inter-process locks """
+
+    homepage = "http://www.python.org/pypi/zc.lockfile"
+    url      = "https://pypi.io/packages/source/z/zc.lockfile/zc.lockfile-1.4.tar.gz"
+
+    version(
+        '1.4', sha256='95a8e3846937ab2991b61703d6e0251d5abb9604e18412e2714e1b90db173253')
+
+    depends_on('py-setuptools', type=('build', 'run'))
+    depends_on('py-zc-buildout', type=('build', 'run'))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-zc-lockfile/package.py]: This dependency isn't listed in the `setup.py`, remove it

Comment (marcmengel) [reply var/spack/repos/builtin/packages/py-zc-lockfile/package.py]: Yes, it's not listed, but It's used in their bootstrap.py...

Comment (adamjstewart) [reply var/spack/repos/builtin/packages/py-zc-lockfile/package.py]: What is this `bootstrap.py` file used for? Is it something that gets installed? Maybe we could make it a `run` type dependency only.

Comment (marcmengel) [reply var/spack/repos/builtin/packages/py-zc-lockfile/package.py]: Okay, well I could have sworn it wouldn't install unless I added the py-zc-buildout package, but I just installed it from the recipe with the dependency removed and its fine without it.  Sorry for the confusion.

Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-zc-lockfile/package.py]: Ping @marcmengel, just need to remove this dependency.

**Comment by adamjstewart** [comment]
Ping ping @marcmengel

**Pull request merged**